pkgs/servers/search/opensearch/default.nix

```
@@ -0,0 +1,46 @@
+{ lib
+, stdenvNoCC
+, fetchurl
+, makeWrapper
+, jre_headless
+, util-linux
+, gnugrep
+, coreutils
+, autoPatchelfHook
+, zlib
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "opensearch";
+  version = "2.3.0";
+
+  src = fetchurl {
+    url = "https://artifacts.opensearch.org/releases/bundle/opensearch/2.3.0/opensearch-2.3.0-linux-x64.tar.gz";
```

```suggestion
  version = "2.3.0";

  src = fetchurl {
    url = "https://artifacts.opensearch.org/releases/bundle/opensearch/${version}/opensearch-${version}-linux-x64.tar.gz";
```

Tie the source URL to the version, so we don't have to modify it on every update. That being said, the latest release seems to be 2.5.0.

And what about other targets than x86_64-linux, are there any?

---

We use the JDK from nix. So it makes no difference for any platform. I have it running on MacOS :)